src/NHibernate.Test/NHSpecificTest/GH1170/Fixture.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Linq;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH1170
+{
+	[TestFixture]
+	public class Fixture : BugTestCase
+	{
+		// Only the set case is tested, because other cases were not affected:
+		// - bags delete everything first.
+		// - indexed collections use their index, which is currently not mappable as a composite index with nullable
+		//   column. All index columns are forced to not-nullable by mapping implementation. When using a formula in
+		//   index, they use the element, but its columns are also forced to not-nullable.
```

Comment:
This "forced to not-nullable" thing seems to me a bit undue, as this impacts only ddl generation, forbidding to test such cases (unless tweaking the generated schema afterward), but not forbidding users not using schema generation to map such things, and insert null. Then these cases will have the delete issue too, and also an issue on update. But no one seems to ask for nullable support on those cases, so I have not handled them.